workspaces/src/error/mod.rs

```diff
@@ -0,0 +1,52 @@
+//! All errors that can occur within workspaces, including but not limited to
+//! the following: IO, RPC, parsing, and serialization errors.
+
+mod parse;
+mod rpc;
+
+pub use self::parse::{ParseError, ParseErrorKind};
+pub use self::rpc::{RpcError, RpcErrorKind};
+
+/// Error type that workspaces will make use of for all the errors
+/// returned from this library
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum Error {
+    #[error("RPC errored out: {0}")]
+    RpcError(#[from] RpcError),
+    #[error("Execution error: {0}")]
+    ExecutionError(String),
+    #[error("sandbox has already been started")]
+    SandboxAlreadyStarted,
+    #[error("sandbox failed to patch state: {0}")]
+    SandboxPatchStateFailure(String),
+    #[error("sandbox failed to fast forward: {0}")]
+    SandboxFastForwardFailure(String),
+    #[error("sandbox failed due to: {0}")]
+    SandboxUnknownError(String),
+    #[error("IO error from {0}")]
+    IoError(#[from] std::io::Error),
+    #[error("account error from {0}")]
+    AccountError(String),
+    #[error("parse error from {0}")]
+    ParseError(#[from] ParseError),
+    #[error("bytes error from {0}")]
+    BytesError(#[from] BytesError),
+    #[error(transparent)]
+    Other(#[from] Box<dyn std::error::Error>),
+}
```

Comment:
What is the motivation for *not* making this an opaque error types? `SandboxUnknownError(String),` and `Other(#[from] Box<dyn std::error::Error>)` as a publicly visible API strike me as highly unexpected. 

I would expect something shaped roughtly like reqwest error: 

https://docs.rs/reqwest/latest/reqwest/struct.Error.html

Reply:
pretty much in the general comment of https://github.com/near/workspaces-rs/pull/149#issuecomment-1156908342 where we have too many different sets of errors that is not intuitive to work with if we just muddle the waters here, IMO